### ISSUE TITLE
Softmax fusion perf regression between 5/18 and 6/12

### ISSUE THREAD

**Issue by kevinstephano** [NVIDIA/Fuser#486]
In addition to the block size and register changing, it looks like vectorization is not used in the new version.

5/18 repro (note the `define_constant` to `define_scalar` api changed).

```
import torch
from nvfuser import FusionDefinition, DataType

def nvfuser_fusion_id1(fd : FusionDefinition) -> None :                                                                                      
    T0 = fd.define_tensor(symbolic_sizes=[-1, -1, -1, -1], contiguity=[True, True, True, True], dtype=DataType.Float, is_cpu=False)
    T1 = fd.define_tensor(symbolic_sizes=[1, 1, -1, -1], contiguity=[None, None, True, True], dtype=DataType.Float, is_cpu=False)
    S2 = fd.define_constant(0.125000, dtype=DataType.Double)
    T3 = fd.ops.mul(T0, S2)                               
    T4 = fd.ops.slice(T1, start_indices=[0, 0, 0, 0], end_indices=[1, 1, 128, 128], strides=[1, 1, 1, 1])                                    
    S5 = fd.define_constant(0.00000, dtype=DataType.Double)
    T6 = fd.ops.eq(T4, S5)                                     
    T7 = fd.ops.broadcast_in_dim(T6, output_shape=[16, 16, 128, 128], broadcast_dims=[0, 1, 2, 3])
    S8 = fd.define_constant(float("-inf"), dtype=DataType.Double)
    T9 = fd.ops.where(T7, S8, T3)
    T10 = fd.ops.max(T9, axes=[3], keepdim=False, dtype=DataType.Null) 
    T11 = fd.ops.broadcast_in_dim(T10, output_shape=[16, 16, 128, 1], broadcast_dims=[0, 1, 2])
    T12 = fd.ops.broadcast_in_dim(T11, output_shape=[16, 16, 128, 128], broadcast_dims=[0, 1, 2, 3])
    T13 = fd.ops.sub(T9, T12)                                                                                                                
    T14 = fd.ops.exp(T13)                                                                                                                    
    T15 = fd.ops.sum(T14, axes=[3], keepdim=False, dtype=DataType.Null)                              
    T16 = fd.ops.broadcast_in_dim(T15, output_shape=[16, 16, 128, 1], broadcast_dims=[0, 1, 2])      
    T17 = fd.ops.broadcast_in_dim(T16, output_shape=[16, 16, 128, 128], broadcast_dims=[0, 1, 2, 3])
    T18 = fd.ops.div(T14, T17)                           
    S19 = fd.define_constant(0.00000, dtype=DataType.Double)
    S20 = fd.define_constant(1.00000, dtype=DataType.Double)
    S21 = fd.define_constant(16, dtype=DataType.Int) 
    S22 = fd.define_constant(16, dtype=DataType.Int)                                                                                           
    S23 = fd.define_constant(128, dtype=DataType.Int)        
    S24 = fd.define_constant(128, dtype=DataType.Int)
    T25 = fd.ops.uniform(S19, S20, shape=[S21, S22, S23, S24], dtype=DataType.Float)
    S26 = fd.define_constant(0.900000, dtype=DataType.Double)
    T27 = fd.ops.lt(T25, S26)
    T28 = fd.ops.cast(T27, dtype=DataType.Float)
    T29 = fd.ops.mul(T18, T28)
    S30 = fd.define_constant(1.11111, dtype=DataType.Double)
    T31 = fd.ops.mul(T29, S30)
    fd.add_output(T31)

with FusionDefinition() as fd:
    nvfuser_fusion_id1(fd)

inputs = [
    torch.randn(16, 16, 128, 128, device='cuda'),
    torch.randn(1, 1, 128, 128, device='cuda'),
]

for _ in range(5):
    out = fd.execute(inputs)
```
The 6/12 repro:

```
import torch
from nvfuser import FusionDefinition, DataType

def nvfuser_fusion_id1(fd : FusionDefinition) -> None :                                                                                      
    T0 = fd.define_tensor(symbolic_sizes=[-1, -1, -1, -1], contiguity=[True, True, True, True], dtype=DataType.Float, is_cpu=False)
    T1 = fd.define_tensor(symbolic_sizes=[1, 1, -1, -1], contiguity=[None, None, True, True], dtype=DataType.Float, is_cpu=False)
    S2 = fd.define_scalar(0.125000, dtype=DataType.Double)
    T3 = fd.ops.mul(T0, S2)                               
    T4 = fd.ops.slice(T1, start_indices=[0, 0, 0, 0], end_indices=[1, 1, 128, 128], strides=[1, 1, 1, 1])                                    
    S5 = fd.define_scalar(0.00000, dtype=DataType.Double)
    T6 = fd.ops.eq(T4, S5)                                     
    T7 = fd.ops.broadcast_in_dim(T6, output_shape=[16, 16, 128, 128], broadcast_dims=[0, 1, 2, 3])
    S8 = fd.define_scalar(float("-inf"), dtype=DataType.Double)
    T9 = fd.ops.where(T7, S8, T3)
    T10 = fd.ops.max(T9, axes=[3], keepdim=False, dtype=DataType.Null) 
    T11 = fd.ops.broadcast_in_dim(T10, output_shape=[16, 16, 128, 1], broadcast_dims=[0, 1, 2])
    T12 = fd.ops.broadcast_in_dim(T11, output_shape=[16, 16, 128, 128], broadcast_dims=[0, 1, 2, 3])
    T13 = fd.ops.sub(T9, T12)                                                                                                                
    T14 = fd.ops.exp(T13)                                                                                                                    
    T15 = fd.ops.sum(T14, axes=[3], keepdim=False, dtype=DataType.Null)                              
    T16 = fd.ops.broadcast_in_dim(T15, output_shape=[16, 16, 128, 1], broadcast_dims=[0, 1, 2])      
    T17 = fd.ops.broadcast_in_dim(T16, output_shape=[16, 16, 128, 128], broadcast_dims=[0, 1, 2, 3])
    T18 = fd.ops.div(T14, T17)                           
    S19 = fd.define_scalar(0.00000, dtype=DataType.Double)
    S20 = fd.define_scalar(1.00000, dtype=DataType.Double)
    S21 = fd.define_scalar(16, dtype=DataType.Int) 
    S22 = fd.define_scalar(16, dtype=DataType.Int)                                                                                           
    S23 = fd.define_scalar(128, dtype=DataType.Int)        
    S24 = fd.define_scalar(128, dtype=DataType.Int)
    T25 = fd.ops.uniform(S19, S20, shape=[S21, S22, S23, S24], dtype=DataType.Float)
    S26 = fd.define_scalar(0.900000, dtype=DataType.Double)
    T27 = fd.ops.lt(T25, S26)
    T28 = fd.ops.cast(T27, dtype=DataType.Float)
    T29 = fd.ops.mul(T18, T28)
    S30 = fd.define_scalar(1.11111, dtype=DataType.Double)
    T31 = fd.ops.mul(T29, S30)
    fd.add_output(T31)

with FusionDefinition() as fd:
    nvfuser_fusion_id1(fd)

inputs = [
    torch.randn(16, 16, 128, 128, device='cuda'),
    torch.randn(1, 1, 128, 128, device='cuda'),
]

for _ in range(5):
    out = fd.execute(inputs)
```

5/18 perf:
```
1869465398          31456     303  32768     1     1    32     1     1       41         0.000         0.000                                                     NVIDIA A100 80GB PCIe (0)    1     7  CudaCodeGen::kernel1

===== Reduction Stats ========
total_reduction_numel: 128                    
total_iteration_numel: 32768                                    
inner_most_dimension_numel: 128
vectorize_factor: 4                                                                                                                          
n_tensor_inputs: 2                                          
max_input_dtype_size: 4                                        
max_persistent_buffer_size: 512          
max_multi_reduction_factor: 256                                                                                                              
block(32, 1, 1)                            
===== Reduction Parameters ========                     
Tag: Inner Persistent Heuristic.    
                                                                                                                                             
Red On Fastest Dim                  
Persistent Kernel                                                                                                                                                                                                                                                                          
Batches per block: 1                                                                                                                         
                                                                                                                                             
Iteration Domain: blockIdx.x /                              
Inner Reduction Domain: cross block - threadIdx.x /  pad to warp / persistent batch - 1 / vectorize / factor 4
Launch Parameters: BlockDim.x = -1, BlockDim.y = 1, BlockDim.z = -1, GridDim.x = -1, GridDim.y = -1, GridDim.z = -1, Smem Size = 0                                                                                                                                                         
                                                                      
```

6/12 perf:
```
1701752207          76513     306  32768     1     1   128     1     1       22         0.000         0.001                                                     NVIDIA A100 80GB PCIe (0)    1     7  CudaCodeGen::kernel1


===== Reduction Stats ========                
total_reduction_numel: 128
total_iteration_numel: 32768                                                                                                                                                                                                                                                               
inner_most_dimension_numel: 128                                                                                                                                                                                                                                                            
vectorize_factor: 1                                                                                                                          
n_tensor_inputs: 2                                                                                                                           
max_input_dtype_size: 4                                   
max_persistent_buffer_size: 512                                                                                                                                                                                                                                                            
max_multi_reduction_factor: 256                          
block(128, 1, 1)                                               
===== Reduction Parameters ========                                                                                                          
Tag: Inner Persistent Heuristic.                               
                                                                      
Red On Fastest Dim                                                                                                                           
Persistent Kernel                                                                                                                            
Batches per block: 1                                                                                                                         
                                                                                                                                                                                                                                                                                           
Iteration Domain: blockIdx.x /                                                                                                                                                                                                                                                             
Inner Reduction Domain: cross block - threadIdx.x /  pad to warp / persistent batch - 1 /                                                    
Launch Parameters: BlockDim.x = -1, BlockDim.y = 1, BlockDim.z = -1, GridDim.x = -1, GridDim.y = -1, GridDim.z = -1, Smem Size = 0           
```

**Comment by kevinstephano** [comment]
From @zasdfgbnm :

> Oh, I see. Good finding, and thank you for looking into it! I do not expect max_supported_vector_size to be 1. Looks like this is a preexisting bug in the system, and 88a782b67c40c5cdb2e8cd5cd2b38311853e99a3 exposes this bug. I will look into it.
